src/sentry/api/endpoints/project_group_index.py

```diff
@@ -132,12 +133,30 @@ def validate_inRelease(self, attrs, source):
                 )
         return attrs
 
+    def validate_inCommit(self, attrs, source):
+        value = attrs[source]
+        project = self.context['project']
+        try:
+            attrs[source] = Commit.objects.get(
```

Comment:
one decision we should make: should this require an explicit repository? if so how should we pass that?

same thing should end up in the response

Reply:
Rough idea:

```
{
  "statusDetails": {
    "inCommit": {
        "id": "sha",
        "repository": "getsentry/sentry"
    }
}
```

Reply:
Do you mean "explicit repository" as in just specifying the repo by name or referencing a `Repository` record we already have associated with the Org?

If we require it to reference a `Repository`, we're cutting off anyone who doesn't use our Integrations, no? (since the only way to get `Repository` records now is through an Integration). You could theoretically use all this today without using Integrations, I think.

Reply:
@mnoble it'd still not work without using integrations, but we'll likely fix that case later by letting you specify a "custom" repository (where you can just funnel commits to our API)

Reply:
I've made the change to be commit+repository in the payload, which resembles how we do it elsewhere.